src/components/scenes/ChangePasswordScene.tsx

```diff
@@ -73,7 +73,15 @@ const ChangePasswordSceneComponent = ({
       return
     }
 
-    return onSubmit(password)
+    setSpinning(true)
+
+    try {
+      onSubmit(password)
+    } catch (e) {
+      showError(e)
+    } finally {
+      setSpinning(!spinning)
```

Comment:
I hate to be pedantic but this should be setSpinning(false). It doesn't matter here, but in general for this pattern:

turn on activity indicator -> do some work -> turn off activity indicator 

We want to be sure to actually turn it off. In more complex code maybe `spinning` is changed in the try block or function calls within it. To be safe, when we get to the finally block we want to be 100% that the activity is turned off by setting to false.